src/lib/gundb/UserStorage.js

```diff
@@ -0,0 +1,44 @@
+//@flow
+import { default as goodWallet, GoodWallet } from '../wallet/GoodWallet'
+import logger from '../logger/pino-logger'
+import initGunDB from './gundb'
+
+export type GunDBUser = {
+  alias: string,
+  epub: string,
+  pub: string
+}
+class UserStorage {
+  wallet: GoodWallet
+  profile: any
+  user: GunDBUser
+
+  constructor() {
+    this.wallet = goodWallet
+    this.wallet.ready.then(() => this.init())
+  }
+
+  async init() {
+    logger.level = 'trace'
+    logger.info('Initializing GunDB UserStorage')
+    initGunDB()
+    const username = await this.wallet.sign('GoodDollarUser')
+    const password = await this.wallet.sign('GoodDollarPass')
+    const gunuser = global.gun.user()
+    gunuser.create(username, password, async res => {
```

Comment:
This is not always working for me. Seems like callback way does.

```
gunuser.create(username, password, res => {
      logger.info('GunDB created', res, username, password)

      gunuser.auth(username, password, res => {
        if (res.err) {
          logger.error("GunDB can't login!", res.err)
          return
        }
        this.user = res
        this.profile = gunuser.get('profile')
        logger.info('GunDB logged in', {
          profile: this.profile,
          user: this.user,
          username,
          pubkey: this.wallet.account
        })
      })
    })
```

Reply:
You mean the promise doesn't resolve?

Reply:
Yes, seems like it works once when the user is being created the promise is resolved
But after refreshing wasn't resolving. Then again changing username and password worked once.
Not sure why this is happening but notice that in this tutorial is being used with the callback

https://github.com/amark/gun/wiki/auth

Reply:
For me its working.
For some reason the default pino logger doesnt work correctly.
I've added: const logger = pino.child({ from: 'UserStorage' })
and now logging works
@fernandomg can you review this? and confirm the promise is being resolved (reaching log "GunDB logged in")

Reply:
It's working fine for me.